>/home/circleci/project/src/server/shared/enuminfo_SharedDefines.cpp:1286:14: fatal error: use of undeclared identifier 'SPELL_ATTR7_UNK27'
        case SPELL_ATTR7_UNK27: return 27;

there is one more location where SPELL_ATTR7_UNK27 was used

--- a/src/server/game/Handlers/SpellHandler.cpp
+++ b/src/server/game/Handlers/SpellHandler.cpp
@@ -591,14 +591,14 @@ void WorldSession::HandleSelfResOpcode(WorldPacket & /*recvData*/)
 {
     TC_LOG_DEBUG("network", "WORLD: CMSG_SELF_RES");                  // empty opcode
 
-	if (uint32 spell = sSpellMgr->GetSpellInfo(_player->GetUInt32Value(PLAYER_SELF_RES_SPELL)))
-	{
-		if (_player->HasAuraType(SPELL_AURA_PREVENT_RESURRECTION) && !spell->HasAttribute(SPELL_ATTR7_BYPASS_PREVENT_RES))
-			return; // silent return, client should display error by itself and not send this opcode
+    if (uint32 spell = sSpellMgr->GetSpellInfo(_player->GetUInt32Value(PLAYER_SELF_RES_SPELL)))
+    {
+        if (_player->HasAuraType(SPELL_AURA_PREVENT_RESURRECTION) && !spell->HasAttribute(SPELL_ATTR7_BYPASS_PREVENT_RES))
+            return; // silent return, client should display error by itself and not send this opcode
 
-		_player->CastSpell(_player, spellId);
-		_player->SetUInt32Value(PLAYER_SELF_RES_SPELL, 0);
-	}
+        _player->CastSpell(_player, spellId);
+        _player->SetUInt32Value(PLAYER_SELF_RES_SPELL, 0);
+    }
 }
 
 void WorldSession::HandleSpellClick(WorldPacket& recvData)

--- a/src/server/game/Handlers/SpellHandler.cpp
+++ b/src/server/game/Handlers/SpellHandler.cpp
@@ -591,7 +591,7 @@ void WorldSession::HandleSelfResOpcode(WorldPacket & /*recvData*/)
 {
     TC_LOG_DEBUG("network", "WORLD: CMSG_SELF_RES");                  // empty opcode
 
-    if (uint32 spell = sSpellMgr->GetSpellInfo(_player->GetUInt32Value(PLAYER_SELF_RES_SPELL)))
+    if (SpellInfo* spell = sSpellMgr->GetSpellInfo(_player->GetUInt32Value(PLAYER_SELF_RES_SPELL)))
     {
         if (_player->HasAuraType(SPELL_AURA_PREVENT_RESURRECTION) && !spell->HasAttribute(SPELL_ATTR7_BYPASS_PREVENT_RES))
             return; // silent return, client should display error by itself and not send this opcode

--- a/src/server/game/Handlers/SpellHandler.cpp
+++ b/src/server/game/Handlers/SpellHandler.cpp
@@ -596,7 +596,7 @@ void WorldSession::HandleSelfResOpcode(WorldPacket & /*recvData*/)
         if (_player->HasAuraType(SPELL_AURA_PREVENT_RESURRECTION) && !spell->HasAttribute(SPELL_ATTR7_BYPASS_PREVENT_RES))
             return; // silent return, client should display error by itself and not send this opcode
 
-        _player->CastSpell(_player, spellId);
+        _player->CastSpell(_player, spell->Id);
         _player->SetUInt32Value(PLAYER_SELF_RES_SPELL, 0);
     }
 }

--- a/src/server/shared/enuminfo_SharedDefines.cpp
+++ b/src/server/shared/enuminfo_SharedDefines.cpp
@@ -1283,7 +1283,7 @@ TC_API_EXPORT size_t EnumUtils<SpellAttr7>::ToIndex(SpellAttr7 value)
         case SPELL_ATTR7_CANT_PARRY: return 24;
         case SPELL_ATTR7_CANT_MISS: return 25;
         case SPELL_ATTR7_UNK26: return 26;
-        case SPELL_ATTR7_UNK27: return 27;
+        case SPELL_ATTR7_BYPASS_PREVENT_RES: return 27;
         case SPELL_ATTR7_CONSOLIDATED_RAID_BUFF: return 28;
         case SPELL_ATTR7_UNK29: return 29;
         case SPELL_ATTR7_UNK30: return 30;

